python/lsst/daf/butler/core/ddl.py

```diff
@@ -170,15 +170,25 @@ def process_bind_param(self, value, dialect):
             _LOG.warning("%s is later than max. time %s, max. time time will be used instead",
                          value, MAX_TIME)
             value = MAX_TIME
-        value = round((value - EPOCH).to_value("sec") * 1e9)
-        return int(value)
+
+        delta = value - EPOCH
+        # Special care needed to preserve nanosecond precision.
+        # Usually jd1 has no fractional part but just in case.
+        jd1, extra_jd2 = divmod(delta.jd1, 1)
+        nsec_per_day = 1_000_000_000 * 24 * 3600
```

Comment:
Can we pull this out as a file constant rather than defining it in both routines?

Reply:
Moved everything to a separate module instead 🙂